firmware/Core/Src/telecommands/telecommand_definitions.c

```diff
@@ -317,23 +309,82 @@ uint8_t TCMDEXEC_set_int_config_variable(const uint8_t *args_str, TCMD_Telecomma
         return 0;
     }
 
-    const uint8_t set_result = CONFIG_assign_int_new_value(&res, config_var_new_value, (uint8_t)config_var_width);
+    const uint8_t set_result = CONFIG_assign_int_new_value(&res, config_var_new_value);
 
     if (set_result > 0)
     {
         snprintf(response_output_buf, response_output_buf_len, "%s : %s\n", "Could not set config var: ", config_var_name);
         return 0;
     }
 
-    snprintf(response_output_buf, response_output_buf_len, "Set config var: %s to " "%" PRIu64 " with width \n", res->variable_name, *(uint64_t*)res->variable_pointer, res->width_bytes);
+    const uint8_t width = res->width_bytes;
+    uint64_t print_val = 0;
+    switch (width)
+    {
+        case 1:
+            print_val = *(uint8_t *)(res->variable_pointer);
+            break;
+        case 2:
+            print_val = *(uint16_t *)(res->variable_pointer);
+            break;
+        case 4:
+            print_val = *(uint32_t *)(res->variable_pointer);
+            break;
+        case 8:
+            print_val = *(uint64_t *)(res->variable_pointer);
+            break;
+    }
+    snprintf(response_output_buf, response_output_buf_len, "Set config var: %s to 0x%016llx with width: %d \n", res->variable_name,print_val , res->width_bytes);
+    // snprintf(response_output_buf, response_output_buf_len, "Width: %d", res->width_bytes);
     return 0;
 }
 
 uint8_t TCMDEXEC_set_str_config_variable(const uint8_t *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
                                          char *response_output_buf, uint16_t response_output_buf_len)
 {
-    // TODO
-    snprintf(response_output_buf, response_output_buf_len, "TODO\n");
+    const int args_str_len = strlen((char*)args_str);
```

Comment:
Please rebase to `main`, and then remove all of these `(char*)args_str` type casts. `args_str` is now a `char*` to follow the convention (because it's a cstring).